psh/tools/psh.py

```diff
@@ -119,6 +124,68 @@ def assert_cmd_successed(pexpect_proc):
     assert _get_exit_code(pexpect_proc) == 0, 'The exit status of last passed command does not equal 0!'
 
 
+def ls(pexpect_proc, dir=''):
+    ''' Returns the list with named tuples containing information about files present in the specified directory '''
+    File = namedtuple('File', ['name', 'owner', 'is_dir', 'date'])
+    Date = namedtuple('Date', ['month', 'mday', 'time'])
+
+    pexpect_proc.sendline(f'ls -la {dir}')
+    pexpect_proc.expect_exact('ls -la')
+    if dir:
+        pexpect_proc.expect_exact(f' {dir}')
+
+    pexpect_proc.expect_exact('\n')
+
+    files = []
+    while True:
+        idx = pexpect_proc.expect([r'\(psh\)\% ', r'([^\r\n]+(\r+\n))'])
+        if idx == 0:
+            break
+
+        line = pexpect_proc.match.group(0)
+        # temporary solution to match line with missing `---` in owner position (issue #254)
+        if '    ---' in line:
+            line = line.replace('    ---', '--- ---')
+        try:
+            permissions, _, owner, _, _, month, mday, time, name = line.split()
+        except ValueError:
+            assert False, f'wrong ls output: {line}'
+
+        # Name is printed with ascii escape characters - remove them
+        esc_sequences = re.findall(CONTROL_CODE, name)
+
+        for seq in esc_sequences:
+            name = name.replace(seq, '')
+
+        f = File(name, owner, permissions[0] == 'd', Date(month, mday, time))
+        files.append(f)
+
+    return files
+
+
+def ls_simple(pexpect_proc, dir=''):
+    ''' Returns list of file names from the specified directory'''
+    files = []
+    pexpect_proc.sendline(f'ls -1 {dir}')
+    idx = pexpect_proc.expect_exact([f'ls -1 {dir}', pexpect.TIMEOUT, pexpect.EOF])
+    assert idx == 0, f"ls {dir} command hasn't been sent properly"
```

Comment:
Asserting EOF and TMEOUT causes that pexpect exception is not raised and we have our assertion message plus reader buffer printed like that:
![Screenshot from 2022-11-21 11-54-26](https://user-images.githubusercontent.com/77304431/203032941-ccb19a12-ad8e-4eaa-82ba-54b01af63e09.png)

It is not crucial, but I think it could be used both for `ls` and `ls_simple`, so I've created _send() function (will be pushed in incoming commit):

```python
def _send(pexpect_proc, cmd):
    pexpect_proc.sendline(cmd)
    idx = pexpect_proc.expect_exact([cmd, pexpect.TIMEOUT, pexpect.EOF])
    assert idx == 0, f"{cmd} command hasn't been sent properly"
```

That's important to assert the whole pattern with prompt in `assert_cmd`, so I would leave it as it is.